addons/mail/static/src/new/core/channel_member_model.js

```diff
@@ -5,10 +5,25 @@
  * @typedef Data
  * @property {number} id
  * @property {number} partnerId
+ * @property {import("@mail/new/core/thread_model").Thread} memberThread
  */
 export class ChannelMember {
-    constructor(_state, { id, partnerId }) {
-        Object.assign(this, { _state, id, partnerId });
+    static insert(state, data) {
+        let channelMember = state.channelMembers[data.id];
+        if (!channelMember) {
+            channelMember = new ChannelMember();
+            state.channelMembers[data.id] = channelMember;
+        }
+        Object.assign(channelMember, {
```

Comment:
All the fields could be defined at the top of the class at well to help with documentation and autocomplete.